src/clusterfuzz/_internal/platforms/android/util.py

```diff
@@ -37,3 +37,44 @@ def get_local_path(device_path):
   return os.path.join(
       root_directory,
       os.path.relpath(device_path, android.constants.DEVICE_FUZZING_DIR))
+
+
+def is_tescase_deprecated(platform_id=None):
+  """Whether or not the Android device is deprecated."""
+
+  # Platform ID for Android is of the form - |android:{codename}_{sanitizer}:{build_version}|
+  platform_id_fields = platform_id.split(':')
+  if len(platform_id_fields) != 3:
+    return False
+
+  codename_fields = platform_id_fields[1].split('_')
+  if len(codename_fields) != 2:
+    return False
+
+  # Check if devices is deprecated
+  if codename_fields[0] in android.constants.DEPRECATED_DEVICE_LIST:
+    return True
+
+  # Check if version is deprecated
+  # Currently only "main" or "m" is active.
+  # All other branches including "master" have been deprecated
+  if (platform_id_fields[2] <= 'v' or platform_id_fields[2] == 'master') and platform_id_fields[2] != 'm'
```

Comment:
I don't think this code will run, can you compare strings using the `<=` operator?

Reply:
This is working as intended.

Reply:
Marking this as resolved